src/main/java/com/synopsys/integration/jenkins/scan/service/ScanCommandsFactory.java

```diff
@@ -42,6 +46,72 @@ public static ScanPipelineCommands createPipelineCommand(Run<?, ?> run, TaskList
+            prepareCoverityParametersMap(scanStep, parametersMap);
+        } else if (scanType.equals(ScanType.POLARIS)) {
+            // preparePolarisParametersMap
+        } else {
+            prepareBlackDuckParametersMap(scanStep, parametersMap);
+        }
+
+        if (!Utility.isStringNullOrBlank(scanStep.getBitbucket_token())) {
+            parametersMap.put(ApplicationConstants.BITBUCKET_TOKEN_KEY, scanStep.getBitbucket_token());
+        }
+
+        prepareBridgeParametersMap(scanStep, parametersMap);
+
+        return parametersMap;
+    }
+
+    private static void setGlobalConfigurationValues(Map<String, Object> parametersMap) {
+        ScannerGlobalConfig config = GlobalConfiguration.all().get(ScannerGlobalConfig.class);
+        if (config != null) {
+            ScanType scanType = ScanType.BLACKDUCK;
+
+            if (!Utility.isStringNullOrBlank(config.getScanType())) {
+                scanType = ScanType.valueOf(config.getScanType());
+                parametersMap.put(ApplicationConstants.SCAN_TYPE_KEY, config.getScanType());
+            }
+
+            if (scanType.equals(ScanType.COVERITY)) {
+                if (!Utility.isStringNullOrBlank(config.getCoverityConnectUrl())) {
+                    parametersMap.put(ApplicationConstants.COVERITY_CONNECT_URL_KEY, config.getCoverityConnectUrl());
+                }
+                if (!Utility.isStringNullOrBlank(config.getCoverityConnectUserName())) {
+                    parametersMap.put(ApplicationConstants.COVERITY_CONNECT_USER_NAME_KEY, config.getCoverityConnectUserName());
+                }
+                if (!Utility.isStringNullOrBlank(config.getCoverityConnectUserPassword())) {
+                    parametersMap.put(ApplicationConstants.COVERITY_CONNECT_USER_PASSWORD_KEY, config.getCoverityConnectUserPassword());
+                }
+            } else if (scanType.equals(ScanType.POLARIS)) {
+                // set polaris global config values
+            } else {
+                if (!Utility.isStringNullOrBlank(config.getBlackDuckUrl())) {
+                    parametersMap.put(ApplicationConstants.BLACKDUCK_URL_KEY, config.getBlackDuckUrl());
+                }
+                if (!Utility.isStringNullOrBlank(config.getBlackDuckApiToken())) {
+                    parametersMap.put(ApplicationConstants.BLACKDUCK_API_TOKEN_KEY, config.getBlackDuckApiToken());
+                }
+            }
+
+            if (!Utility.isStringNullOrBlank(config.getBitbucketToken())) {
+                parametersMap.put(ApplicationConstants.BITBUCKET_TOKEN_KEY, config.getBitbucketToken());
+            }
+
+            if (!Utility.isStringNullOrBlank(config.getSynopsysBridgeDownloadUrl())) {
+                parametersMap.put(ApplicationConstants.BRIDGE_DOWNLOAD_URL, config.getBlackDuckApiToken());
+            }
+        }
+    }
+
+    private static void prepareBlackDuckParametersMap(SecurityScanStep scanStep, Map<String, Object> parametersMap) {
```

Comment:
Better approach might be returning a new map, rather then mutating the method parameter.

Reply:
Updated accordingly